ci-operator/step-registry/cucushift/upgrade/check/capability/cucushift-upgrade-check-capability-commands.sh

```diff
@@ -0,0 +1,439 @@
+    # # Split the capability names based on spaces and '+' signs
+    # known_caps=$(echo "${tmp_known_caps[*]}" | tr ' ' '\n' | tr '+' '\n' | sort -u | uniq)
+    # echo "Known capabilities list is ${known_caps}"
+}
+
+# Get implicitly enabled capabilities from cluster
+function get_actual_implicit_caps {
+    echo "Getting actual implicitly enabled caps in cluster..."
+    local tmp_capabilities implicit_status implicit_message
+    # Obtain ImplicitlyEnabledCapabilities condition status from cluster
+    if ! implicit_status=$(oc get clusterversion version -o json | jq -r '.status.conditions[] | select(.type == "ImplicitlyEnabledCapabilities").status'); then
+        echo "ERROR: Get implicitly enabled caps failed"
+        return 1
+    fi
+    if [ "$implicit_status" = "False" ]; then
+        echo "No implicitly enabled capabilities in the cluster"
+        return
+    fi
+    # Obtain ImplicitlyEnabledCapabilities condition message from cluster
+    if ! implicit_message=$(oc get clusterversion version -o json | jq -r '.status.conditions[] | select(.type == "ImplicitlyEnabledCapabilities").message'); then
+        echo "ERROR: Get implicitly enabled caps failed"
+        return 1
+    fi
+
+    # Parse the message to extract implicit caps
+    # The condition message could be "The following capabilities could not be disabled: Console, Insights, Storage"
+    tmp_capabilities=$(echo "$implicit_message" | grep -oE 'could not be disabled: (.*)' | cut -d':' -f2)
+    actual_implicit_caps=$(echo "$tmp_capabilities" | awk -F', ' '{ for (i=1; i<=NF; i++) print $i }')
+    echo "Actual implicitly enabled capabilities list is ${actual_implicit_caps}"
+}
+
+# Get expected implicit enabled capabilities via comparision between caps_set variables
+# It doesn't take migrated caps into account
+function get_new_caps() {
+    echo "Getting new caps introduced in target release as compared to source release..."
+    local source="$1"
+    local target="$2"
+    local array_source array_target extra_caps
+
+    # Check for missing environment variables
+    if [ -z "$source" ] || [ -z "$target" ]; then
+        echo "ERROR: Missing required arguments"
+        return 1
+    fi
+
+    # Split the elements into arrays
+    IFS=" " read -ra array_source <<< "${caps_set[$source]}"
+    IFS=" " read -ra array_target <<< "${caps_set[$target]}"
+
+    # Find the extra elements in caps_set[$target] compared to caps_set[$source]
+    extra_caps=()
+    for cap in "${array_target[@]}"; do
+    #shellcheck disable=SC2076
+    if [[ ! " ${array_source[*]} " =~ " ${cap} " ]]; then
+        extra_caps+=("$cap")
+    fi
+    done
+
+    # Combine the extra elements into a space-separated string
+    expected_implicit_caps="${extra_caps[*]}"
+    echo "Expected implicitly enabled capabilities list is ${expected_implicit_caps}"
+}
+
+# compares two payloads for capabilities difference at manifest level
+function extract_migrated_caps() {
```

Comment:
Shall we run `oc` commands to check the enabled resources instead of checking the extracted manifest?

Reply:
Ignore the above comments, maybe I mis-understand the case.